_sources/UsingRESTAPIs/flickr.rst

```diff
@@ -28,88 +28,71 @@ The structure of a URL for a photo search on flickr is:
 
 Let's put everything together to make a little retrieval tool for flickr images containing particular tags. Of course, in a browser, you can just use flickr's search tool. But doing this through the API opens up other possibilities that you can explore for features not provided on the regular flickr website.
 
-Below is some code that queries the flickr API for images that have a particular tag (I have found that searching for "mountains", "Switzerland", and "cows" usually produces beautiful images that are "safe for work", so the example below does that search.)
-
-.. note:
-
-    To run this code, you will need to copy it to a file on your local machine (not an activecode window), and **paste in an api_key that you get from flickr**.
+Below is some code that queries the flickr API for images that have a particular tag (I have found that searching for "mountains" and "rivers" usually produces beautiful images that are "safe for work", so the example below does that search.)
 
 .. sourcecode:: python
 
+    # import statements
     import requests
     import json
-    import pickle
     import webbrowser
-
-    def canonical_order(d):
```

Comment:
Why did you remove the caching from here? Is it before caching gets introduced? If it's after, why not keep using caching, to reinforce it?